### PR TITLE
Feature/refactoring broadcast

### PR DESCRIPTION
**Pull request by warbhe-amit**
code refactoring wrt sendEvent and sendBroadcastEvent

### CODE REVIEW

File: server/src/util.mjs [hunk 1]
```diff
@@ -45,6 +47,52 @@ function createTmpFile(prefix, postfix) {
...
+  logger.info(
+    `Could not send ${onMethod} event because no listener is active`
+  );
+};
+
+function fFatalErr() {
+  logger.info(`Internal error`);
+};
+
+function sendEvent(ws, userId, onMethod, result, msg) {
```
Comment (michaelcfine) [inline server/src/util.mjs:66]: I see what you're doing here (and with sendBroadcastEvent, but I don't think there should be a dependency from util.mjs to any other file; I'd like to keep it for independent "generic utilities". But you can put these functions inside events.mjs if you want, or just go back to where you're passing in the function pointers, but just pass util.fSuccess, etc.

**Review by michaelcfine — CHANGES_REQUESTED** [review]
Looks good, but I'd like to keep util.mjs free from any dependencies into 'real' code. See comment on util.mjs; either move the utility functions into events.mjs and continue to use or go back to using the full signatures and passing the function pointers, just pass util.fSuccess, etc. (instead of the local anonymous definitions, which you've replaced nicely).

File: server/src/util.mjs (hunk 1 above)
Comment (warbhe-amit) [reply server/src/util.mjs]: changes done, please check and confirm

**Review by michaelcfine — DISMISSED** [review]
👍 Looks good to me. Thanks for keeping util.mjs nice and simple / dependency-free!

**Comment by warbhe-amit** [comment]
@michaelcfine, rebased with main. Kindly approve it again in order to merge it.

**Review by michaelcfine — DISMISSED** [review]
👍 FWIW, I really don't like the partial imports and then not prefixing the calls (e.g., events.sendEvent vs. sendEvent), but let's leave it as-is.

**Comment by warbhe-amit** [comment]
rebasing with main requires approval

File: server/src/events.mjs [hunk 2]
```diff
@@ -24,9 +24,25 @@ import * as stateManagement from './stateManagement.mjs';
...
 import { config } from './config.mjs';
 
+function logSuccess(onMethod, result, msg) {
+  logger.info(
+    `${msg}: Sent event ${onMethod} with result ${JSON.stringify(result)}`
+  );
+};
+
+function logErr(onMethod) {
+  logger.info(
```
Comment (vyvanand) [inline server/src/events.mjs:37]: can this be changed to logger.error?

File: server/src/events.mjs [hunk 3]
```diff
@@ -24,9 +24,25 @@ import * as stateManagement from './stateManagement.mjs';
...
+    `Could not send ${onMethod} event because no listener is active`
+  );
+};
+
+function logFatalErr() {
+  logger.info(`Internal error`);
+};
+
 // Maps full userIds to maps which map event listner request method
 // name (e.g., lifecycle.onInactive) to message id (e.g., 17)
```
Comment (vyvanand) [inline server/src/events.mjs:47]: can this be changed to logger.error?

File: server/src/events.mjs (hunk 2 above)
Comment (warbhe-amit) [reply server/src/events.mjs:37]: Earlier also this was logger.info only, so if we want to change then should be checked with Mike.

File: server/src/events.mjs (hunk 3 above)
Comment (warbhe-amit) [reply server/src/events.mjs:47]: Earlier also this was logger.info only, so if we want to change then should be checked with Mike.

File: server/src/events.mjs [hunk 4]
```diff
@@ -160,28 +176,28 @@ function coreSendEvent(isBroadcast, ws, userId, method, result, msg, fSuccess, f
...
-                  logger.info(`${msg}: Sent event ${method} with result ${JSON.stringify(result)}`)
-                }
-                function fErr() {
-                  logger.info(`Could not send ${method} event because no listener is active`)
-                }
-                function fFatalErr() {
-                  logger.info(`Internal error`)
-                }
-                sendEvent(ws, userId, method, result, msg, fSuccess, fErr, fFatalErr);
+                sendEvent(
```
Comment (kschrief) [inline server/src/events.mjs:179]: What's the reason for splitting this up into multiple lines?

File: server/src/stateManagement.mjs [hunk 5]
```diff
@@ -130,6 +130,23 @@ function handleSequenceOfResponseValues(userId, methodName, params, resp, userSt
   return resp;
 }
 
+// Log Error for invalid methodName
+function logError(methodName, resultErrors, resp) {
```
Comment (kschrief) [inline server/src/stateManagement.mjs:134]: Can we change this name to be more specific to what it's logging?

File: server/src/triggers.mjs [hunk 6]
```diff
@@ -27,6 +27,31 @@ import * as commandLine from './commandLine.mjs';
 const methodTriggers = {};
 const eventTriggers = {};
 
+// Log error
+function logError(errorType, pathDetails, ex) {
```
Comment (kschrief) [inline server/src/triggers.mjs:31]: Can we change this name to be more specific to what it's logging?

**Review by kschrief — CHANGES_REQUESTED** [review]

File: server/src/events.mjs (hunk 4 above)
Comment (warbhe-amit) [reply server/src/events.mjs]: Taken care

File: server/src/triggers.mjs (hunk 6 above)
Comment (warbhe-amit) [reply server/src/triggers.mjs]: Taken care

File: server/src/stateManagement.mjs (hunk 5 above)
Comment (warbhe-amit) [reply server/src/stateManagement.mjs]: Taken care

**Review by kschrief — APPROVED** [review]

**Review by dhillomk — APPROVED** [review]

**Pull request merged**